lib/routes/devices-sessions.js

```diff
@@ -41,33 +38,33 @@ module.exports = (log, db, config, customs, push, devices) => {
         validate: {
           payload: isA.alternatives().try(
             isA.object({
-              id: isA.string().length(32).regex(HEX_STRING).required(),
-              name: isA.string().max(255).regex(DISPLAY_SAFE_UNICODE).optional(),
-              type: isA.string().max(16).optional(),
-              pushCallback: isA.string().uri({ scheme: 'https' }).regex(PUSH_SERVER_REGEX).max(255).optional().allow(''),
-              pushPublicKey: isA.string().max(88).regex(URL_SAFE_BASE_64).optional().allow(''),
-              pushAuthKey: isA.string().max(24).regex(URL_SAFE_BASE_64).optional().allow('')
+              id: DEVICES_SCHEMA.id.required(),
+              name: DEVICES_SCHEMA.name.optional(),
+              type: DEVICES_SCHEMA.type.optional(),
+              pushCallback: DEVICES_SCHEMA.pushCallback.optional(),
+              pushPublicKey: DEVICES_SCHEMA.pushPublicKey.optional(),
+              pushAuthKey: DEVICES_SCHEMA.pushAuthKey.optional()
             }).or('name', 'type', 'pushCallback', 'pushPublicKey', 'pushAuthKey').and('pushPublicKey', 'pushAuthKey'),
             isA.object({
-              name: isA.string().max(255).regex(DISPLAY_SAFE_UNICODE).required(),
-              type: isA.string().max(16).required(),
-              pushCallback: isA.string().uri({ scheme: 'https' }).regex(PUSH_SERVER_REGEX).max(255).optional().allow(''),
-              pushPublicKey: isA.string().max(88).regex(URL_SAFE_BASE_64).optional().allow(''),
-              pushAuthKey: isA.string().max(24).regex(URL_SAFE_BASE_64).optional().allow('')
+              name: DEVICES_SCHEMA.name.required(),
+              type: DEVICES_SCHEMA.type.required(),
+              pushCallback: DEVICES_SCHEMA.pushCallback.optional(),
+              pushPublicKey: DEVICES_SCHEMA.pushPublicKey.optional(),
+              pushAuthKey: DEVICES_SCHEMA.pushAuthKey.optional()
             }).and('pushPublicKey', 'pushAuthKey')
           )
         },
         response: {
           schema: isA.object({
-            id: isA.string().length(32).regex(HEX_STRING).required(),
+            id: DEVICES_SCHEMA.id.required(),
             createdAt: isA.number().positive().optional(),
             // We previously allowed devices to register with arbitrary unicode names,
             // so we can't assert DISPLAY_SAFE_UNICODE in the response schema.
-            name: isA.string().max(255).optional(),
-            type: isA.string().max(16).optional(),
-            pushCallback: isA.string().uri({ scheme: 'https' }).max(255).optional().allow(''),
-            pushPublicKey: isA.string().max(88).regex(URL_SAFE_BASE_64).optional().allow(''),
-            pushAuthKey: isA.string().max(24).regex(URL_SAFE_BASE_64).optional().allow('')
+            name: DEVICES_SCHEMA.name.optional(),
```

Comment:
The comment here implies that it shouldn't enforce the `DISPLAY_SAFE_UNICODE` on response, but the regex `DEVICES_SCHEMA.name` does enforce it. Is it safe to use this or should comment be removed?

Reply:
Whoops, I was a bit trigger happy with the the `CMD+V` on this one. Well spotted, thanks!

Reply:
@vbudhram, what do you reckon if I add a `DEVICES_SCHEMA.nameResponse` property and use that here and for the other responses? Yay or nay?

Reply:
YaY!

Reply:
...i.e. something like:

```
diff --git a/lib/devices.js b/lib/devices.js
index 36371429..3442bc95 100644
--- a/lib/devices.js
+++ b/lib/devices.js
@@ -15,6 +15,9 @@ const PUSH_SERVER_REGEX = require('../config').get('push.allowedServerRegex')
 const SCHEMA = {
   id: isA.string().length(32).regex(HEX_STRING),
   name: isA.string().max(255).regex(DISPLAY_SAFE_UNICODE),
+  // We previously allowed devices to register with arbitrary unicode names,
+  // so we can't assert DISPLAY_SAFE_UNICODE in the response schema.
+  nameResponse: isA.string().max(255),
   type: isA.string().max(16),
   pushCallback: isA.string().uri({ scheme: 'https' }).regex(PUSH_SERVER_REGEX).max(255).allow(''),
   pushPublicKey: isA.string().max(88).regex(URL_SAFE_BASE_64).allow(''),
diff --git a/lib/routes/devices-sessions.js b/lib/routes/devices-sessions.js
index 1fedb1f0..96133e80 100644
--- a/lib/routes/devices-sessions.js
+++ b/lib/routes/devices-sessions.js
@@ -58,9 +58,7 @@ module.exports = (log, db, config, customs, push, devices) => {
           schema: isA.object({
             id: DEVICES_SCHEMA.id.required(),
             createdAt: isA.number().positive().optional(),
-            // We previously allowed devices to register with arbitrary unicode names,
-            // so we can't assert DISPLAY_SAFE_UNICODE in the response schema.
-            name: isA.string().max(255).optional(),
+            name: DEVICES_SCHEMA.nameResponse.optional(),
             type: DEVICES_SCHEMA.type.optional(),
             pushCallback: DEVICES_SCHEMA.pushCallback.optional(),
             pushPublicKey: DEVICES_SCHEMA.pushPublicKey.optional(),
@@ -254,9 +252,7 @@ module.exports = (log, db, config, customs, push, devices) => {
             isCurrentDevice: isA.boolean().required(),
             lastAccessTime: isA.number().min(0).required().allow(null),
             lastAccessTimeFormatted: isA.string().optional().allow(''),
-            // We previously allowed devices to register with arbitrary unicode names,
-            // so we can't assert DISPLAY_SAFE_UNICODE in the response schema.
-            name: isA.string().max(255).required().allow(''),
+            name: DEVICES_SCHEMA.nameResponse.allow('').required(),
             type: DEVICES_SCHEMA.type.required(),
             pushCallback: DEVICES_SCHEMA.pushCallback.allow(null).optional(),
             pushPublicKey: DEVICES_SCHEMA.pushPublicKey.allow(null).optional(),
@@ -314,9 +310,7 @@ module.exports = (log, db, config, customs, push, devices) => {
             userAgent: isA.string().max(255).required().allow(''),
             os: isA.string().max(255).allow('').allow(null),
             deviceId: DEVICES_SCHEMA.id.allow(null).required(),
-            // We previously allowed devices to register with arbitrary unicode names,
-            // so we can't assert DISPLAY_SAFE_UNICODE in the response schema.
-            deviceName: isA.string().max(255).required().allow('').allow(null),
+            deviceName: DEVICES_SCHEMA.nameResponse.allow('').allow(null).required(),
             deviceType: DEVICES_SCHEMA.type.allow(null).required(),
             deviceCallbackURL: DEVICES_SCHEMA.pushCallback.allow(null).required(),
             deviceCallbackPublicKey: DEVICES_SCHEMA.pushPublicKey.allow(null).required(),
```